spd/models/component_model.py

```diff
@@ -283,33 +284,57 @@ def from_pretrained(cls, path: ModelPath) -> tuple["ComponentModel", Config, Pat
         comp_model.load_state_dict(model_weights)
         return comp_model, config, out_dir
 
+    def calc_causal_importances(
+        self,
+        pre_weight_acts: dict[str, Tensor],
+        detach_inputs: bool = False,
+        sigmoid_type: SigmoidTypes = "leaky_hard",
+    ) -> tuple[dict[str, Float[Tensor, "... C"]], dict[str, Float[Tensor, "... C"]]]:
+        """Calculate component activations and causal importances in one pass to save memory.
```

Comment:
Sorry this "component activations" part is unnecessary (I probably forgot to remove it fro man AI output). Can you remove?